File: support_tools/swigwin-3.0.12/Examples/test-suite/go/cpp_namespace_runme.go
// Note: This example assumes that namespaces are flattened
package main

import "./cpp_namespace"

func main() {
	n := cpp_namespace.Fact(4)
	if n != 24 {
		panic("Bad return value!")
	}

	if cpp_namespace.GetFoo() != 42 {
		panic("Bad variable value!")
	}

	t := cpp_namespace.NewTest()
	if t.Method() != "Test::method" {
		panic("Bad method return value!")
	}

	if cpp_namespace.Do_method(t) != "Test::method" {
		panic("Bad return value!")
	}

	if cpp_namespace.Do_method2(t) != "Test::method" {
		panic("Bad return value!")
	}

	cpp_namespace.Weird("hello", 4)

	cpp_namespace.DeleteTest(t)

	t2 := cpp_namespace.NewTest2()
	t3 := cpp_namespace.NewTest3()
	t4 := cpp_namespace.NewTest4()
	t5 := cpp_namespace.NewTest5()

	if cpp_namespace.Foo3(42) != 42 {
		panic("Bad return value!")
	}

	if cpp_namespace.Do_method3(t2, 40) != "Test2::method" {
		panic("Bad return value!")
	}

	if cpp_namespace.Do_method3(t3, 40) != "Test3::method" {
		panic("Bad return value!")
	}

	if cpp_namespace.Do_method3(t4, 40) != "Test4::method" {
		panic("Bad return value!")
	}

	if cpp_namespace.Do_method3(t5, 40) != "Test5::method" {
		panic("Bad return value!")
	}
}

File: support_tools/swigwin-3.0.12/Examples/test-suite/python/cpp_namespace_runme.py
# Note: This example assumes that namespaces are flattened
import cpp_namespace

n = cpp_namespace.fact(4)
if n != 24:
    raise RuntimeError("Bad return value!")

if cpp_namespace.cvar.Foo != 42:
    raise RuntimeError("Bad variable value!")

t = cpp_namespace.Test()
if t.method() != "Test::method":
    raise RuntimeError("Bad method return value!")

if cpp_namespace.do_method(t) != "Test::method":
    raise RuntimeError("Bad return value!")

if cpp_namespace.do_method2(t) != "Test::method":
    raise RuntimeError("Bad return value!")

cpp_namespace.weird("hello", 4)

del t

t2 = cpp_namespace.Test2()
t3 = cpp_namespace.Test3()
t4 = cpp_namespace.Test4()
t5 = cpp_namespace.Test5()

if cpp_namespace.foo3(42) != 42:
    raise RuntimeError("Bad return value!")

if cpp_namespace.do_method3(t2, 40) != "Test2::method":
    raise RuntimeError("Bad return value!")

if cpp_namespace.do_method3(t3, 40) != "Test3::method":
    raise RuntimeError("Bad return value!")

if cpp_namespace.do_method3(t4, 40) != "Test4::method":
    raise RuntimeError("Bad return value!")

if cpp_namespace.do_method3(t5, 40) != "Test5::method":
    raise RuntimeError("Bad return value!")
